src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -84,10 +103,18 @@ internal class PackageRestoreDataSource : ChainedProjectValueDataSourceBase<Rest
             _fileSystem = fileSystem;
             _logger = logger;
             _projectDependentFileChangeNotificationService = projectDependentFileChangeNotificationService;
+            _solutionRestoreService4 = solutionRestoreService4;
         }
 
         protected override IDisposable? LinkExternalInput(ITargetBlock<IProjectVersionedValue<RestoreData>> targetBlock)
         {
+            // Register before this project receives any data flows containing possible nominations.
+            // This is needed because we need to register before any nuget restore or before the solution load.
+            _project.Services.FaultHandler.Forget(
+                _solutionRestoreService4.RegisterRestoreInfoSourceAsync(this, _projectAsynchronousTasksService.UnloadCancellationToken), 
```

Comment:
Can we wrap RegisterRestoreInfoSourceAsync inside a Task.Run, that will ensure exception will be handled by the fault service, instead of breaking this component.